### PR TITLE
header_rewrite: Adds basic string concatenation

### PR DESCRIPTION
**Pull request by randall**
eg:

add-header X-Party "let's party like it's " + %{NOW:YEAR} + "!"

### CODE REVIEW

File: plugins/header_rewrite/conditions.cc [hunk 1]
```diff
@@ -1373,3 +1374,50 @@ ConditionInbound::append_value(std::string &s, const Resources &res, NetworkSess
...
+
+bool
+ConditionStringLiteral::eval(const Resources &res)
+{
+  std::string s;
+
+  append_value(s, res);
+  TSDebug(PLUGIN_NAME, "Evaluating StringLiteral");
+
+  return static_cast<const MatcherType *>(_matcher)->test(s);
```
Comment (zwoop) [inline plugins/header_rewrite/conditions.cc]: Why does it have to copy this string? Couldn't the matcher test() directly on _literal ?

File: doc/admin-guide/plugins/header_rewrite.en.rst [hunk 2]
```diff
@@ -804,11 +804,15 @@ L      Last rule, do not continue.
 QSA    Append the results of the rule to the query string.
 ====== ========================================================================
 
-Variable Expansion
-------------------
+Values and Variable Expansion
+-----------------------------
+You can concatenate values using strings, condition values and variable expansions via the ``+`` operator. The elements in the concatenation must be must be enclosed in double quotes ``"``::
```
Comment (maskit) [inline doc/admin-guide/plugins/header_rewrite.en.rst]: "must be must be" must be "must be".

Comment (randall) [reply doc/admin-guide/plugins/header_rewrite.en.rst]: This "must be" fixed now.

File: doc/admin-guide/plugins/header_rewrite.en.rst [hunk 3]
```diff
@@ -804,11 +804,15 @@ L      Last rule, do not continue.
 QSA    Append the results of the rule to the query string.
 ====== ========================================================================
 
-Variable Expansion
-------------------
+Values and Variable Expansion
+-----------------------------
+You can concatenate values using strings, condition values and variable expansions via the ``+`` operator. The elements in the concatenation must be enclosed in double quotes ``"``::
+
+    add-header CustomHeader "Hello from " + "%{IP:SERVER}" + ":" + "%<INBOUND:LOCAL-PORT>"
```
Comment (bryancall) [inline doc/admin-guide/plugins/header_rewrite.en.rst]: Can you change the documentation to not have the quotes for the condition expansion (e.g. %{IP:SERVER} vs "%{IP:SERVER}")?  I am assuming the quotes are not required and are optional.

Comment (randall) [reply doc/admin-guide/plugins/header_rewrite.en.rst]: Not for %{tag} style tags. They're required for %<> style tags. 

I'll push an update for the docs.

Comment (bryancall) [reply doc/admin-guide/plugins/header_rewrite.en.rst]: Sounds like %<> are going away for 9.0.0 - so I am OK with it

**Comment by bryancall** [comment]
Cherry picked to 8.1.x branch

**Pull request merged**